app/controllers/helpdocs_controller.rb

```diff
@@ -1,83 +1,31 @@
 class HelpdocsController < ApplicationController
-  # GET /helpdocs
-  # GET /helpdocs.json
+
   def index
-    @helpdocs = Helpdoc.all
+    dir = Rails.root.join('app', 'views', 'helpdocs')
+    @files = (
+      (Dir.entries(dir).select{|f| f.match(/^\w.*\.md*/)}
+      .map{|f| f.gsub(/\..*/, '')}) - ['index']
+    ).sort
+    # todo eventually use `tree` or something to display a
+    # tree of available helpdocs. BUT #show doesn't currently
+    # support a tree of them yet.
 
-    respond_to do |format|
-      format.html # index.html.erb
-      format.json { render json: @helpdocs }
-    end
   end
-
   # GET /helpdocs/1
-  # GET /helpdocs/1.json
   def show
-    @helpdoc = Helpdoc.find(params[:id])
-
-    respond_to do |format|
-      format.html # show.html.erb
-      format.json { render json: @helpdoc }
+    if params[:id] == 'index'
+      redirect_to action: 'index', status: :moved_permanently
+      return
     end
-  end
-
-  # GET /helpdocs/new
-  # GET /helpdocs/new.json
-  def new
-    @helpdoc = Helpdoc.new
-
-    respond_to do |format|
-      format.html # new.html.erb
-      format.json { render json: @helpdoc }
-    end
-  end
-
-  # GET /helpdocs/1/edit
-  def edit
-    @helpdoc = Helpdoc.find(params[:id])
-  end
-
-  # POST /helpdocs
-  # POST /helpdocs.json
-  def create
-    @helpdoc = Helpdoc.new(params[:helpdoc])
-
-    respond_to do |format|
-      if @helpdoc.save
-        format.html { redirect_to @helpdoc, notice: 'Helpdoc was successfully created.' }
-        format.json { render json: @helpdoc, status: :created, location: @helpdoc }
-      else
-        format.html { render action: "new" }
-        format.json { render json: @helpdoc.errors, status: :unprocessable_entity }
-      end
-    end
-  end
-
-  # PUT /helpdocs/1
-  # PUT /helpdocs/1.json
-  def update
-    @helpdoc = Helpdoc.find(params[:id])
-
-    respond_to do |format|
-      if @helpdoc.update_attributes(params[:helpdoc])
-        format.html { redirect_to @helpdoc, notice: 'Helpdoc was successfully updated.' }
-        format.json { head :no_content }
-      else
-        format.html { render action: "edit" }
-        format.json { render json: @helpdoc.errors, status: :unprocessable_entity }
+    file_name = "#{File.basename(params[:id])}.md"
+    path = Rails.root.join('app', 'views', 'helpdocs', file_name)
+    if File.exist? path
+      respond_to do |format|
+        format.html {
+          render file_name
```

Comment:
it works the same way as if it were a haml file. You create a template handler and register it, tell it what extensions it supports ( see `config/initializers/markdown_handler.rb`)  and the render method figures it out from there. What I can't figure out why (in this code) `.md` works but `.html.md.erb` does not. I thought I'd be able to just register a handler for the pipeline and it would handle things one extension at a time.... but no. If i want it to handle the ERB code in it at the same time it has to be `.md` not `.md.erb` and i just have to override the `erb` method in my handler. Doesn't seem right but...